main.js

```diff
@@ -44,127 +44,134 @@ it is available then var.The variable is available inside a block and it's child
 You can define the same variable name inside a child without produce a conflict with the variable of the parent block.*/
 let variable = 1234;
 
-// create adapter instance wich will be used for communication with controller
-let adapter;
-function startAdapter(options) {
-	options = options || {};
-	Object.assign(options, {
-        // name has to be set and has to be equal to adapters folder name and main file name excluding extension
-        name: adapterName,
-        // is called when adapter shuts down - callback has to be called under any circumstances!
-        unload: function (callback) {
-            try {
-                adapter.log.info('cleaned everything up...');
-                callback();
-            } catch (e) {
-                callback();
+// define adapter class wich will be used for communication with controller
+class MyAdapter extends utils.Adapter {
+    constructor(options) {
+        let adapterOptions = {
+            // name has to be set and has to be equal to adapters folder name and main file name excluding extension
+            name: adapterName,
+
+            // is called when adapter shuts down - callback has to be called under any circumstances!
+            unload: this._unload,
```

Comment:
I would bind those to the current `this` context: `this._unload.bind(this)`. Since we don't know in which context the function is invoked from js-controller, this is safer:

See also https://reactjs.org/docs/handling-events.html for an example where this is necessary.